pyomo/contrib/solver/results.py

```diff
@@ -33,107 +33,94 @@ class TerminationCondition(enum.Enum):
 
     maxTimeLimit = 1
+    """The solver exited due to reaching a specified time limit."""
 
     iterationLimit = 2
+    """The solver exited due to reaching a specified iteration limit."""
 
     objectiveLimit = 3
+    """The solver exited due to reaching an objective limit. For example, in
+    Gurobi, the exit message "Optimal objective for model was proven to
+    be worse than the value specified in the Cutoff parameter" would map
+    to objectiveLimit.
+    """
 
     minStepLength = 4
+    """The solver exited due to a minimum step length.  Minimum step length
+    reached may mean that the problem is infeasible or that the problem
+    is feasible but the solver could not converge.
+    """
 
     unbounded = 5
+    "The solver exited because the problem has been found to be unbounded."
 
     provenInfeasible = 6
+    "The solver exited because the problem has been proven infeasible."
 
     locallyInfeasible = 7
+    """The solver exited because no feasible solution was found to the
+    submitted problem, but it could not be proven that no such solution
+    exists.
+    """
 
     infeasibleOrUnbounded = 8
+    """Some solvers do not specify between infeasibility or unboundedness
+    and instead return that one or the other has occurred. For example,
+    in Gurobi, this may occur because there are some steps in presolve
+    that prevent Gurobi from distinguishing between infeasibility and
+    unboundedness.
+    """
 
     error = 9
+    """The solver exited with some error. The error message will also be
+    captured and returned.
+    """
 
     interrupted = 10
+    "The solver was interrupted while running."
 
     licensingProblems = 11
+    """The solver experienced issues with licensing. This could be that no
+    license was found, the license is of the wrong type for the problem
+    (e.g., problem is too big for type of license), or there was an
+    issue contacting a licensing server.
+    """
 
     emptyModel = 12
+    "The model being solved did not have any variables"
 
     unknown = 42
+    "All other unrecognized exit statuses fall in this category."
 
 
 class SolutionStatus(enum.Enum):
-    """
-    An enumeration for interpreting the result of a termination. This describes the designated
-    status by the solver to be loaded back into the model.
+    """An enumeration for interpreting the result of a termination. This
+    describes the designated status by the solver to be loaded back into
+    the model.
 
-    Attributes
-    ----------
-    noSolution: 0
-        No (single) solution was found; possible that a population of solutions
-        was returned.
-    infeasible: 10
-        Solution point does not satisfy some domains and/or constraints.
-    feasible: 20
-        A solution for which all of the constraints in the model are satisfied.
-    optimal: 30
-        A feasible solution where the objective function reaches its specified
-        sense (e.g., maximum, minimum)
```

Comment:
Ditto here - can these be put back so the documentation is easier for folks?

Reply:
I would really rather not move the documentation into the class docstring for several reasons:
1. it requires us to duplicate information (the enum value is manually written into the class docstring, and there is no obvious way to write a test to ensure that the documentation stays up-to-date).
2. it leads to the following Sphinx warnings:
```
[pyomo]/pyomo/contrib/solver/results.py:docstring of pyomo.contrib.solver.results.SolutionStatus.noSolution:1: WARNING: duplicate object description of pyomo.contrib.solver.results.SolutionStatus.noSolution, other instance in api/pyomo.contrib.solver.results.SolutionStatus, use :no-index: for one of them
[pyomo]/pyomo/contrib/solver/results.py:docstring of pyomo.contrib.solver.results.SolutionStatus.infeasible:1: WARNING: duplicate object description of pyomo.contrib.solver.results.SolutionStatus.infeasible, other instance in api/pyomo.contrib.solver.results.SolutionStatus, use :no-index: for one of them
[pyomo]/pyomo/contrib/solver/results.py:docstring of pyomo.contrib.solver.results.SolutionStatus.feasible:1: WARNING: duplicate object description of pyomo.contrib.solver.results.SolutionStatus.feasible, other instance in api/pyomo.contrib.solver.results.SolutionStatus, use :no-index: for one of them
[pyomo]/pyomo/contrib/solver/results.py:docstring of pyomo.contrib.solver.results.SolutionStatus.optimal:1: WARNING: duplicate object description of pyomo.contrib.solver.results.SolutionStatus.optimal, other instance in api/pyomo.contrib.solver.results.SolutionStatus, use :no-index: for one of them
```
3. The online docs *don't* pick up that those are actually documenting the Enum members.  You get this:
![image](https://github.com/user-attachments/assets/7af9952a-b3d7-40b6-b615-16178a0df11f)
instead of
![image](https://github.com/user-attachments/assets/92cf4622-fd96-43f1-85e7-1d661ddbadc7)

4. While you `help()` won't show the attribute descriptions (because Python doesn't allow attributes to have `__doc__` members), I would personally prefer that given decision that forces us to do a trade between the (searchable, indexed, and browsable) online docs and the interactive `help()`, I would lean into the online docs.  For the record, the `help()` looks like this if the attributes are documented in the class docstring:
```
>>> from pyomo.contrib.solver.results import SolutionStatus
>>> help(SolutionStatus)
class SolutionStatus(enum.Enum)
 |  SolutionStatus(value, names=None, *, module=None, qualname=None, type=None, start=1, boundary=None)
 |
 |  An enumeration for interpreting the result of a termination. This
 |  describes the designated status by the solver to be loaded back into
 |  the model.
 |
 |  Attributes
 |  ----------
 |  noSolution: 0
 |      No (single) solution was found; possible that a population of solutions
 |      was returned.
 |  infeasible: 10
 |      Solution point does not satisfy some domains and/or constraints.
 |  feasible: 20
 |      A solution for which all of the constraints in the model are satisfied.
 |  optimal: 30
 |      A feasible solution where the objective function reaches its specified
 |      sense (e.g., maximum, minimum)
 |
 |  Method resolution order:
 |      SolutionStatus
 |      enum.Enum
 |      builtins.object
 |
 |  Data and other attributes defined here:
 |
 |  feasible = <SolutionStatus.feasible: 20>
 |
 |  infeasible = <SolutionStatus.infeasible: 10>
 |
 |  noSolution = <SolutionStatus.noSolution: 0>
 |
 |  optimal = <SolutionStatus.optimal: 30>
 |
```
and when only the attributes are documented (using PEP257), you get 
```
>>> from pyomo.contrib.solver.results import SolutionStatus
>>> help(SolutionStatus)
class SolutionStatus(enum.Enum)
 |  SolutionStatus(value, names=None, *, module=None, qualname=None, type=None, start=1, boundary=None)
 |
 |  An enumeration for interpreting the result of a termination. This
 |  describes the designated status by the solver to be loaded back into
 |  the model.
 |
 |  Method resolution order:
 |      SolutionStatus
 |      enum.Enum
 |      builtins.object
 |
 |  Data and other attributes defined here:
 |
 |  feasible = <SolutionStatus.feasible: 20>
 |
 |  infeasible = <SolutionStatus.infeasible: 10>
 |
 |  noSolution = <SolutionStatus.noSolution: 0>
 |
 |  optimal = <SolutionStatus.optimal: 30>
 |
```

Reply:
Okay, I am comfortable with this change then